extends Node

@onready var pause_menu = %PauseMenu

const file_path = "res://Timeline.rk"

func _ready():
	Rakugo.sg_say.connect(_on_say)
	Rakugo.sg_step.connect(_on_step)
	Rakugo.sg_execute_script_finished.connect(_on_execute_script_finished)
  
	Rakugo.parse_and_execute_script(file_path)
  
func _on_say(character:Dictionary, text:String):
	prints("Say", character.get("name", ""), text)
  
func _on_step():
	prints("Press \"Enter\" to continue...")
	
func _on_execute_script_finished(file_name:String, error_str:String):
	prints("End of script")
  
func _process(delta):
	if Rakugo.is_waiting_step() and Input.is_action_just_pressed("ui_accept"):
		Rakugo.do_step()
	if pause_menu.visible == false and Input.is_action_just_pressed("ui_cancel"):
		pause_menu.show()
		pause_menu.set_process(true)
		get_tree().paused = true
